src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py

```diff
@@ -145,3 +162,27 @@ def scan_tokens(self):
             self.process_cred_file(file_name)
 
         super(LocalCredmon, self).scan_tokens
+
+
+def get_user_groups(username, mapfile):
+    """Scan mapfile to find groups for username,
+    returns a list of groups.
+
+    Note that every call will read the current file,
+    regardless of if HTCondor or the credmon have
+    been reconfigured.
+    
+    Mapfiles have the format:
+    * <userA> <group1>,<group2>,...,<groupN>
+    * <userB> <group1>,<group2>,...,<groupN>
+    """
+    groups = []
+    with open(mapfile) as f:
+        for line in f:
+            tokens = line.rstrip().split()
```

Comment:
Why rstrip() instead of strip()?

Reply:
Well, I knew that `split()` ignores some whitespace, but I _thought_ that I still needed to get rid of the line break characters, but apparently line breaks are included in the whitespace characters that `split()` ignores, so we can probably take out `rstrip()` entirely:

```
>>> " a b c d ".split()
['a', 'b', 'c', 'd']
>>> " a b c d \n".split()
['a', 'b', 'c', 'd']
>>> " a b c d \r\n".split()
['a', 'b', 'c', 'd']
```

Reply:
BTW, this reminds me, I should add some checks for if the line doesn't have enough tokens... new commit incoming...

Reply:
@JaimeFrey addressed in https://github.com/htcondor/htcondor/pull/947/commits/5832992f0162f104b8a800a16033dfc8ac2f4dfd

Reply:
Looks good.